django/db/models/query.py

```diff
@@ -151,9 +155,12 @@ class NamedValuesListIterable(ValuesListIterable):
     @staticmethod
     @lru_cache()
     def create_namedtuple_class(*names):
-        # Cache namedtuple() with @lru_cache() since it's too slow to be
+        # Cache type() with @lru_cache() since it's too slow to be
         # called for every QuerySet evaluation.
-        return namedtuple('Row', names)
+        return type('Row', (namedtuple('Row', names),), {
+            '__slots__': (),
+            '__reduce__': lambda self: (_named_row_unpickle, (names, tuple(self)))
+        })
```

Comment:
I would move `create_namedtuple_class` outside of `NamedValuesListIterable` or even better to `django.db.models.utils`, e.g.
```python
def unpickle_row(names, values):
    return create_namedtuple_class(*names)(*values)


@functools.lru_cache()
def create_namedtuple_class(*names):
    # Cache type() with @lru_cache() since it's too slow to be called for every
    # QuerySet evaluation.
    def namedtuple_reduce(self):
        return unpickle_row, (names, tuple(self))

    return type('Row', (namedtuple('Row', names),), {
        '__reduce__': namedtuple_reduce,
    })
```

Reply:
I agree